src/python/foglamp/core/storage_server/storage.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+"""Starts the FogLAMP Storage service"""
+import json
+import os
+import logging
+import signal
+import sys
+import time
+import subprocess
+import requests
+import setproctitle
+
+from foglamp import logger
+from foglamp.core.service_registry.service_registry import Service
+
+__author__ = "Amarendra K Sinha"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+_PID_PATH = os.path.expanduser('~/var/run/storage.pid')
+_WORKING_DIR = os.path.expanduser('~/var/log')
+
+_WAIT_STOP_SECONDS = 5
+"""How many seconds to wait for the core server process to stop"""
+_MAX_STOP_RETRY = 5
+"""How many times to send TERM signal to core server process when stopping"""
+
+
+class Storage:
+    logging_configured = False
+    """Set to true when it's safe to use logging"""
+
+    @classmethod
+    def _configure_logging(cls):
+        """Alters the root logger to send messages to syslog
+           with a filter of WARNING
+        """
+        if cls.logging_configured:
+            return
+
+        logger.setup()
+        cls.logging_configured = True
+
+    @classmethod
+    def run(cls):
+        cls._configure_logging()
+
+        # TODO: Think of some method, other than this, to run the external command within this process, otherwise,
+        # this creates a defunct parent.
+        from subprocess import Popen
```

Comment:
It looks like there are three forks to start the storage service:
server.py multiprocess
start forks once
run creates a subprocess

I do see the defunct process in the process list. It seems like we would only need to spawn once. Am I mistaken?